### PR TITLE
Aggregations: Clean up response API for Aggregations

### PR DESCRIPTION
**Pull request by colings86**
This change makes the response API object for Histogram Aggregations the same for all types of Histogram, and does the same for all types of Ranges.
The change removes getBucketByKey() from all aggregations except filters and terms. It also reduces the methods on the Bucket class to just getKey() and getKeyAsString().
The getKey() method returns Object and the actual Type is returns will be appropriate for the type of aggregation being run. e.g. date_histogram will return a DateTime for this method and Histogram will return a Number.

### CODE REVIEW

File: src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/InternalGeoHashGrid.java [hunk 1]
```diff
@@ -157,7 +156,7 @@ public void writeTo(StreamOutput out) throws IOException {
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
             builder.startObject();
-            builder.field(CommonFields.KEY, getKeyAsText());
+            builder.field(CommonFields.KEY, new StringText(getKeyAsString()));
```
Comment (jpountz) [inline src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/InternalGeoHashGrid.java]: Do we actually need to wrap into a Text object here, it should work with the string?

**Comment by jpountz** [comment]
LGTM

**Comment by colings86** [comment]
@jpountz Pushed a new commit with your suggestion and documentation changes. Could you review the doc changes?

**Comment by jpountz** [comment]
LGTM

**Pull request merged**